### PR TITLE
#297 Typing for typescript

### PR DESCRIPTION
**Pull request by MerlinPerrotLegler**
Hello,

This is probably not the best implementation, usually I use typescript definition auto-generate, but he's lost here :S

To test this, I've use `yarn add https://github.com/MerlinPerrotLegler/opening_hours.js.git#feature/typescript-definition` after having remove 'husky' from the `package.json` (as for some reason it's not working with either yarn or mac).

If I forgot something let me know :)

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+export declare enum warnings_severity {
+  warning = 4,
+  notice = 5,
+  info = 6,
+  debug = 7,
+}
+
+export interface optional_conf {
+  mode: mode
```
Comment (HolgerJeromin) [inline types/index.d.ts:68]: Is here a comma missing?

Comment (MerlinPerrotLegler) [reply types/index.d.ts]: yes, interface can be written witg `,`, `;` or nothing (as enum, as code or as class)...
I've add eslint by default it added `,`.

File: package.json [hunk 2]
```diff
@@ -55,12 +56,14 @@
...
     "@rollup/plugin-node-resolve": "^11.2.1",
     "@rollup/plugin-yaml": "^3.0.0",
+    "@typescript-eslint/eslint-plugin": "^4.26.1",
+    "@typescript-eslint/parser": "^4.26.1",
     "assert": "^2.0.0",
     "colors": "^0.6.2",
     "country-language": "^0.1.7",
     "doctoc": "^2.0.0",
     "esbuild": "^0.11.14",
-    "eslint": "^7.25.0",
```
Comment (MerlinPerrotLegler) [inline package.json:62]: minor update should iI change this ?

File: package.json [hunk 3]
```diff
@@ -34,7 +35,7 @@
     "build": "make build",
     "test-quick": "make check",
     "test": "make check-full",
-    "eslint": "eslint ./src/index.js",
+    "eslint": "eslint src/**/*.js types/*.d.ts",
```
Comment (MerlinPerrotLegler) [inline package.json:38]: make sure it's suits you

File: package.json [hunk 4]
```diff
@@ -24,7 +24,8 @@
     "CHANGELOG.rst",
     "LICENSES/",
     "site/js/",
-    "src/locales/core.js"
+    "src/locales/core.js",
+    "types/index.d.ts"
```
Comment (MerlinPerrotLegler) [inline package.json:28]: required for npm

**Comment by ypid** [comment]
LGTM, thanks! I am not using typescript but the typing definition looks precise to me.

**Pull request merged**